### ISSUE TITLE
LACT AMD GPU  configuration tool overclock not working

### ISSUE THREAD

**Issue by sabo999** [getsolus/packages#4032]
### Please confirm there isn't an existing open bug report

- [x] I have searched open bugs for this issue

### Summary

when you first open LACT there is a sudo command you need to do to enable the daemon. that is fine and that works correctly. the problem comes when you go into the overclocking tab, to be able to change parameters on your gpu you need to press a tab called "overclock" and i dont know what it does but it regenerates initramfs afterwards. The error pops up there
![Screenshot from 2024-10-11 12-33-12](https://github.com/user-attachments/assets/4613b954-6ddf-48b2-83f8-af840bd0095b)

### Steps to reproduce

1. open LACT
2. do sudo command and close LACT
3. reopen LACT and go to OC tab
4. at top press "overclock" button and the error should pop up
5. 

### Expected result

it should say overclocking enabled, please reboot the system to enable, or something similar to this

### Actual result

produced an error and nothing happened, no overclock controls are being enabled even with rebooting

### Environment

- [X] Is system up to date?

### Repo

Shannon (stable)

### Desktop Environment

GNOME

### System details

System:
  Host: ammod-solus Kernel: 6.10.13-304.current arch: x86_64 bits: 64
  Desktop: GNOME v: 46.5 Distro: Solus 4.5 resilience
Machine:
  Type: Desktop Mobo: Gigabyte model: B650 GAMING X AX V2
    serial: <superuser required> UEFI: American Megatrends LLC. v: F32c
    date: 09/27/2024
CPU:
  Info: 8-core AMD Ryzen 7 7800X3D [MT MCP] speed (MHz): avg: 4921
    min/max: 545/5050
Graphics:
  Device-1: Advanced Micro Devices [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900
    XTX/7900 GRE/7900M] driver: amdgpu v: kernel
  Device-2: Logitech Webcam C270 driver: snd-usb-audio,uvcvideo type: USB
  Display: wayland server: X.Org v: 24.1.3 with: Xwayland v: 24.1.3
    compositor: gnome-shell driver: gpu: amdgpu resolution: 2560x1080~200Hz
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.2.3 renderer: AMD
    Radeon RX 7900 XT (radeonsi navi31 LLVM 18.1.8 DRM 3.57
    6.10.13-304.current)
Network:
  Device-1: Realtek RTL8852CE PCIe 802.11ax Wireless Network
    driver: rtw89_8852ce
  Device-2: Realtek RTL8125 2.5GbE driver: r8169
Drives:
  Local Storage: total: 15.24 TiB used: 505.31 GiB (3.2%)
Info:
  Memory: total: 32 GiB note: est. available: 30.96 GiB
    used: 13.72 GiB (44.3%)
  Processes: 447 Uptime: 5h 10m Shell: Bash inxi: 3.3.36

**Comment by uni-dos** [comment]
Do you have the kernel [parameter](https://wiki.archlinux.org/title/AMDGPU#Overclocking) set?

**Comment by sabo999** [comment]
i'm going to be very blunt and honest with you here. I have NEVER EVER ONCE had to do that in ANY other distro, so no, i have no idea how to do that. In any other distro, i just click the button, it does something, and i reboot.  I looked through that document but 1. i dont know which boot system solus uses 2. i dont know where to add the kernel parameter 3. i dont know how to rebuild initramfs, if i even have to.  thanks for your time

**Comment by uni-dos** [comment]
No worries. So for lact to work, it needs to have a boot parameter set. If you type this command into your terminal `printf 'amdgpu.ppfeaturemask=0x%x\n' "$(($(cat /sys/module/amdgpu/parameters/ppfeaturemask) | 0x4000))"` you'll get some amdgpu.ppfeaturemask and some hex values. Next, you'll follow [this](https://help.getsol.us/docs/user/quick-start/boot-management/). But if you do not want to read it, type the following `echo '$outputfromcommand' | sudo tee /etc/kernel/cmdline.d/40_amdgpu_control.conf` Finally, type `sudo clr-boot-manager update` and reboot.

**Comment by malfisya** [comment]
Maybe you can include that file with package? Maybe ship it somewhere in `/usr/lib` ( I am not reading the docs yet). For example `droidcam` need to ship some [config file]( https://github.com/getsolus/packages/blob/373a8d39dfc152203b7f8c6b93f1a559fe8762b9/packages/d/droidcam/package.yml#L41-L42) to work out of the box.

**Comment by sabo999** [comment]
sorry to say but i did it like you said and even read the doc and tried it a few different way but overclocking was not enabled and i was not able to see my clocks or any info at all.

**Comment by ReillyBrogan** [comment]
Alright, after the next sync this should work out of the box after a reboot so long as you're using the -current kernel. Please let me know if it doesn't work.

**Comment by endumiuz** [comment]
> next sync this should work out of the box after a reboot so long as you're using the -current kernel. Please let me know if it do

It doesn't work out of the box for me. I still have to add `amdgpu.ppfeaturemask` to the kernel parameters to get overdrive working.

**Comment by ReillyBrogan** [comment]
Are you using the latest kernel? `6.11.5-307.current`? What is the output of `cat /proc/cmdline`?

**Comment by endumiuz** [comment]
Yes, 6.11.5-307.current

cat /proc/cmdline:
`initrd=\EFI\com.solus-project\initrd-com.solus-project.current.6.11.5-307 root=UUID=1f60ff8a-7a84-4179-a716-645eefa12640 rd.luks.uuid=a0a21140-71fd-4500-abb2-ba5e3f366d9f quiet splash rw amdgpu.overdrive_enabled=1 resume=UUID=9be34d24-d246-4cda-b8bb-e698afe09f2f rd.luks.options=discard`

**Comment by ReillyBrogan** [comment]
Alright, thanks for your patience on this. I checked the LACT source code and it was directly checking the power features flag and wasn't aware that overdrive was enabled despite the flag not matching. I patched it to read the new module parameter and updated the displayed warning to be more applicable to Solus users. I also removed the initramfs generation support since none of that worked on Solus anyway.
